Contract - create a balancer pool that'll be used for testing

bconfortin opened the pull request:
Closes #40

bconfortin committed:
Prettified contract

bconfortin committed:
Prettified contract

bconfortin committed:
Prettified contract

justinbarry committed:
Move new issues to triage column vs backlog

bthaile committed:
fixed getting market shares and lp tokens for the user (#141)

* fixed getting market shares and lp tokens for the user

* fix type errors

justinbarry reviewed (changes requested):
@bconfortin Would you be able to add a test for this to prove it works?

bconfortin commented:
> @bconfortin Would you be able to add a test for this to prove it works?

That was going to be my next task

bthaile committed:
show positions (#147)

JohnDanz committed:
More fixes (#146)

* fixed chart buttons to show pricing for outcomes, fixed approval check hook to update on blocks.

* formatting

bthaile committed:
Fix remove liquidity share balances (#150)

* show positions

* fix removing liqudiity and share balances

* fix link from portfolio page

* fix display user position on market page

* fix merged conflict and format

robert-e-davidson3 committed:
Massive refactor in turbo architecture (#124)

Instead of a single system, there are now several.
The centerpiece is MarketFactory, which creates and resolves
markets. As well as handling shares.

also, using openzepplin erc20


Co-authored-by: robert <robert.davidson@forecastfoundation.org>

robert-e-davidson3 committed:
smart changes for ui integration (#154)

* smart changes for ui integration

* deploy to kovan

Co-authored-by: robert <robert.davidson@forecastfoundation.org>

robert-e-davidson3 commented on packages/smart/contracts/balancer/BPoolForTesting.sol:
```diff
@@ -0,0 +1,29 @@
+        uint256[] calldata _initialLiquidity,
+        uint256[] calldata _weights
+    ) external returns (BPool) {
+        require(_tokens.length == _weights.length);
```
Also verify that the initial liquidities list is the same length, as well as adding a message explaining why the call failed.

robert-e-davidson3 commented on packages/smart/contracts/balancer/BPoolForTesting.sol:
```diff
@@ -0,0 +1,29 @@
+    uint256 private constant MAX_UINT = 2**256 - 1;
+
+    function createBPoolForTesting(
+        ERC20[] calldata _tokens,
```
Ought to be the interface, IERC20, not the implementation.

robert-e-davidson3 commented on packages/smart/contracts/balancer/BPoolForTesting.sol:
```diff
@@ -0,0 +1,29 @@
+pragma solidity 0.7.6;
+
+import "./BFactory.sol";
+import "../libraries/ERC20.sol";
```
This has changed since you started. Now we are use openzepplin's ERC20, from their npm package. Rebase off of dev and there will be examples... it's a small change. As I noted below though, use IERC20 instead of ERC20.

robert-e-davidson3 reviewed (commented):
Some suggestions. Only the updated import is necessary.

bconfortin committed:
Merge branch 'bruno/contract-balancer-pool' of https://github.com/AugurProject/turbo into bruno/contract-balancer-pool

bconfortin committed:
Added Robert's changes from dev

bconfortin committed:
Added Robert's changes from dev

JohnDanz committed:
Teamdata (#155)

* added teams.ts to start testing building team data

* updated teams.ts task, now creating large teams object.

* renamed file, updated hint

* more renaming, going for camelCase instead.

* forgot to save index

* prettier

robert-e-davidson3 committed:
fix verify and redeploy (#161)

Co-authored-by: robert <robert.davidson@forecastfoundation.org>

bthaile committed:
Integrate new market contracts (#159)

* pull new market data format from contracts in multicall

* checking in, issue with getting balances, commented out getting lp token balance

* usdc now is 6 decimals

* formatting

bthaile committed:
remove unused types, add console logs for trading (#162)

* remove unused types, add console logs for trading

* formatting

JohnDanz committed:
Generate get teams (#170)

* updated getTeams, generated first teams.json file for comps.

* added team helpers, added sports.json. generated an updated teams.json to include new sports

* formatting

* exporting in Utils and TeamHelpers to levvel namespace for Comps

JohnDanz committed:
Invalid updates (#174)

* created an orderOutcomesForDisplay helper, applied to market card outcomes table

* updated chart outcome selection buttons to include no contest, no longer explicitly re-ordering outcome 1 and 2

* cleaning/formatting

phoebemirman committed:
fund link task & Chainlink contract (#160)

* add chainlink contract

robert-e-davidson3 committed:
handle sub-18 decimals aka buy shares for less than $100k (#177)

* handle sub-18 decimals aka buy shares for less than $100k


Co-authored-by: robert <robert.davidson@forecastfoundation.org>

bthaile committed:
Use team name data mock market (#176)

* mocked up data for market just to get something to look at

* formatting

* remove price call and calc based on ratios

* formatting

bconfortin committed:
Added tests to the test balancer pool

bconfortin committed:
Added tests to the test balancer pool

bconfortin committed:
Prettified and linted

bconfortin committed:
Prettified and linted

bthaile committed:
sports market cards will be broken up into 3 sections. title descript… (#178)

* sports market cards will be broken up into 3 sections. title description start time

* formatting

JohnDanz committed:
Update categories (#180)

* updated sports.json to include more ccategories, updated logic for category labels to use 3rd category label if available.

* added type ignore for irrelevant check, formatting

bconfortin committed:
Merge branch 'bruno/contract-balancer-pool' of https://github.com/AugurProject/turbo into bruno/contract-balancer-pool

bconfortin commented:
Opened a new, cleaner PR
